### PR TITLE
Upgrade to GHC 8.6

### PR DESCRIPTION
**Pull request by mightybyte**

### CODE REVIEW

File: chainweb.cabal [hunk 1]
```diff
@@ -348,7 +349,7 @@ library
         , swagger2 >= 2.2
         , temporary >= 1.3
         , text >= 1.2
-        , these >= 0.7 && < 1
+        , these >= 0.7 && < 1.1
```
Comment (fosskers) [inline chainweb.cabal]: The `semialign` dep effectively forces `these >= 1`.

File: overrides.nix [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+         }) {};
+
+    ourOverrides = {
+      pact = dontCheck ( addBuildDepend (self.callCabal2nix "pact" pactSrc {}) pkgs.z3);
+
+      aeson = callHackageDirect {
+        pkg = "aeson";
+        ver = "1.4.3.0";
+        sha256 = "13lim8vv78m9lhn7qfjswg7ax825gn0v75gcb80hckxawgk8zxc1";
+      };
```
Comment (fosskers) [inline overrides.nix]: This should be safe to unpin by now.

File: chainweb.cabal (hunk 1 above)
Comment (mightybyte) [reply chainweb.cabal]: Oh, right.  I'll make that change.

File: overrides.nix (hunk 2 above)
Comment (mightybyte) [reply overrides.nix]: It's not safe to unpin.  This nixpkgs comes with aeson-1.4.2.  :(

Comment (fosskers) [reply overrides.nix]: Ah :(

**Pull request merged**